test/Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -80,12 +84,46 @@ spec = Test.group "Meta-Value Manipulation" <|
         Meta.is_an err Error . should_be_true
         Meta.is_a err Text . should_be_false
 
+    Test.specify "should allow for returning the type of a value" <|
+        Meta.type_of 42 . should_equal Integer
+        Meta.type_of 2.81 . should_equal Decimal
+        Meta.type_of 2.81 . should_not_equal Integer
+
+        # Can't compare like
+        # Meta.type_of [1,2,3] . should_equal Vector.Vector
+        # because it tries to compare lengths, where RHS is a type
+        Meta.is_a (Meta.type_of [1,2,3]) Vector.Vector . should_be_true
+
+        Meta.type_of "foo" . should_equal Text
+        Meta.type_of "0" . should_not_equal Integer
+
+        Meta.type_of True . should_equal Boolean
+        Meta.type_of False . should_not_equal Any
+
+        Meta.is_a (Meta.type_of Date.now) Date.Date . should_be_true
```

Comment:
Is this correct check? Will this be correct check in the future?

`Meta.is_a` shall _Check if `self` is an instance of `typ`_. But here it is used to test `==`! `Meta.type_of Date.now` returns `Date.Date`. Right now `Date.Date` may be instance of `Date.Date`, but that's more a bug than a feature - it is certainly not true in Smalltalk - `Class` isn't instance of itself - neither in Java.

Don't you want to replace the check with `should_equal`?

Reply:
Yes, the test is not ideal but I also cannot replace it with `should_equal`
```
    - [FAILED] should allow for returning the type of a value
        Reason: An unexpected panic was thrown: (Unsupported_Argument_Types_Data [Date, Date])
        at <enso> case_branch(.../enso/enso/built-distribution/enso-engine-0.0.0-dev-linux-amd64/enso-0.0.0-dev/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso:358:24-64)
```

Reply:
I decided to implement a separate function `is_of_type` that only deals with comparing types. Using `==` was a hack that _sometimes_ worked by accident, as you noticed, and we should avoid it.